### PR TITLE
Add docs for `.on()`, `.then()`, and `.success()`, as well as the subclasses of `gr.EventData`

### PR DESCRIPTION
**Pull request by abidlabs**
Adds several classes and functions to the "Helpers" section of the documentation.

* Documents `.on()`

Closes: #7976 

* Documents the `Dependency` class with its `.then()` and `success()` methods

Closes: https://github.com/gradio-app/gradio/issues/5072
Closes: https://github.com/gradio-app/gradio/issues/7348

* Documents `gr.EventData` better as well as all of its subclasses explicitly

Closes: https://github.com/gradio-app/gradio/issues/7380

image

See: https://gradio-p2wvb6ymp-hugging-face.vercel.app/main/docs/gradio/likedata

### CODE REVIEW

File: js/_website/src/lib/templates/gradio/04_helpers/eventdata.svx [hunk 1]
```diff
@@ -29,23 +29,27 @@ gradio.EventData(···)
...
+    textbox.select(on_select, textbox, statement)
+
+demo.launch()
 ```
 {/if}
 
 <!--- Initialization -->
-### Initialization
+### Attributes
 <ParamTable parameters={obj.parameters} />
```
Comment (aliabd) [inline js/_website/src/lib/templates/gradio/04_helpers/eventdata.svx:53]: this is coming with 'required' which doesn't make sense here for an attribute. to fix just add  `obj.parameters[0]["kwargs"] = null` in the script section in the top.

**Review by aliabd — COMMENTED** [review]
Thanks @abidlabs for adding all of these docs and closing long standing issues! 

A few notes: 
- you aren't currently documenting `.then` and `.success`. Did you mean to add them manually as attributes to `dependency.svx`? 
- can you use a demo without external dependencies for `LikeData`? `chatbot_multimodal` is breaking for this reason. You can try it out on https://www.gradio.app/playground first 
- since there are a lot of pages in the helpers section now, maybe ordering them makes sense? nbd but for example `DeletedFileData` mentions `EventData` which comes later. If it makes sense to you, you can prefix the important ones with 01_ , 02_ etc. It's a better experience when we highlight the important ones first.

**Comment by abidlabs** [comment]
Thanks @aliabd for the review!

> you aren't currently documenting .then and .success. Did you mean to add them manually as attributes to dependency.svx?

as discussed in our sync just now, I've left them out because they are attributes but also functions, which make them hard to document automatically. instead, I've just described them in the docstring and included a code example for each

> can you use a demo without external dependencies for LikeData? chatbot_multimodal is breaking for this reason. You can try it out on https://www.gradio.app/playground first

will do

> since there are a lot of pages in the helpers section now, maybe ordering them makes sense? nbd but for example DeletedFileData mentions EventData which comes later. If it makes sense to you, you can prefix the important ones with 01_ , 02_ etc. It's a better experience when we highlight the important ones first

did not know, will do!

File: js/_website/src/lib/templates/gradio/04_helpers/01_eventdata.svx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    import {get_object} from "../../process_json.ts";
+    import ParamTable from "$lib/components/ParamTable.svelte";
+    import ShortcutTable from "$lib/components/ShortcutTable.svelte";
+    import DemosSection from "$lib/components/DemosSection.svelte";
+    import FunctionsSection from "$lib/components/FunctionsSection.svelte";
+    import GuidesSection from "$lib/components/GuidesSection.svelte";
+    import CopyButton from "$lib/components/CopyButton.svelte";
+    import { style_formatted_text } from "$lib/text";
+
+    let obj = get_object("eventdata");
```
Comment (aliabd) [inline js/_website/src/lib/templates/gradio/04_helpers/01_eventdata.svx:12]: I think you forgot to add the kwargs thing here 


Screen Shot 2024-06-25 at 7 59 43 PM

Comment (aliabd) [reply js/_website/src/lib/templates/gradio/04_helpers/01_eventdata.svx:12]:
```suggestion
    let obj = get_object("eventdata");
    obj.parameters[0]["kwargs"] = null;
```

**Review by aliabd — APPROVED** [review]
Nice @abidlabs everything looks good apart from one comment!

**Comment by abidlabs** [comment]
Thanks @aliabd for the great review!

**Pull request merged**